test/functional/FunctionalCodeActionsSpec.hs

```diff
@@ -278,6 +278,29 @@ spec = describe "code actions" $ do
             \  where\n\
             \    stuff (A a) = A (a + 1)\n"
 
+  describe "missing top level signature code actions" $
+    it "Adds top level signature" $
+      runSession hieCommand fullCaps "test/testdata/" $ do
+        doc <- openDoc "TopLevelSignature.hs" "haskell"
+
+        _ <- waitForDiagnosticsSource "ghcmod"
+        cas <- map (\(CACodeAction x)-> x) <$> getAllCodeActions doc
```

Comment:
Can you reuse `fromAction` here?
https://github.com/haskell/haskell-ide-engine/blob/deae46c393988872c212e0ab595dbeb28e78bd60/test/functional/FunctionalCodeActionsSpec.hs#L304-L306

Reply:
cool, I'll try that